src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java

```diff
@@ -106,4 +203,95 @@ public ErrorCode getErrorCode() {
   public Response.Status getHttpStatus() {
     return httpStatus;
   }
+
+  private ErrorFamily getErrorFamily() {
+    if (serverFamily.contains(errorCode)
+        || errorCode.name().startsWith("SERVER")
+        || errorCode.name().startsWith("EMBEDDING")) {
+      return ErrorFamily.SERVER;
+    }
+    return ErrorFamily.REQUEST;
+  }
+
+  private ErrorScope getErrorScope(ErrorFamily family) {
+    // first handle special cases
+    if (errorCode == SERVER_INTERNAL_ERROR) {
+      return ErrorScope.EMPTY;
+    }
+    for (Map.Entry<Set<ErrorCode>, ErrorScope> entry : errorCodeScopeMap.entrySet()) {
+      if (entry.getKey().contains(errorCode)) {
+        return entry.getValue();
+      }
+    }
+
+    // decide the scope based in error code pattern
+    if (errorCode.name().contains("SCHEMA")) {
+      return ErrorScope.SCHEMA;
+    }
+    if (errorCode.name().startsWith("EMBEDDING") || errorCode.name().startsWith("VECTORIZE")) {
+      return ErrorScope.EMBEDDING;
+    }
+    if (errorCode.name().contains("FILTER")) {
+      return ErrorScope.FILTER;
+    }
+    if (errorCode.name().contains("SORT")) {
+      return ErrorScope.SORT;
+    }
+    if (errorCode.name().contains("INDEX")) {
+      return ErrorScope.INDEX;
+    }
+    if (errorCode.name().contains("UPDATE")) {
+      return ErrorScope.UPDATE;
+    }
+    if (errorCode.name().contains("SHRED") || errorCode.name().contains("DOCUMENT")) {
+      return ErrorScope.DOCUMENT;
+    }
+    if (errorCode.name().contains("PROJECTION")) {
+      return ErrorScope.PROJECTION;
+    }
+    if (errorCode.name().contains("AUTHENTICATION")) {
+      return ErrorScope.AUTHENTICATION;
+    }
+    if (errorCode.name().contains("OFFLINE")) {
+      return ErrorScope.DATA_LOADER;
+    }
+
+    // decide the scope based on family
+    if (family == ErrorFamily.SERVER) {
+      return ErrorScope.DATABASE;
+    }
+
+    return ErrorScope.EMPTY;
+  }
+
+  enum ErrorFamily {
+    SERVER,
+    REQUEST
+  }
+
+  enum ErrorScope {
+    AUTHENTICATION("AUTHENTICATION"),
+    DATA_LOADER("DATA_LOADER"),
+    DATABASE("DATABASE"),
+    DOCUMENT("DOCUMENT"),
+    EMBEDDING("EMBEDDING"),
+    EMPTY(""),
+    FILTER("FILTER"),
+    INDEX("INDEX"),
+    PROJECTION("PROJECTION"),
+    SCHEMA("SCHEMA"),
+    SORT("SORT"),
+    UPDATE("UPDATE");
```

Comment:
Should there also be an error scope for OPTIONS if you pass in an invalid option?